components/admin_console/feature_discovery/feature_discovery.tsx

```diff
@@ -125,52 +126,49 @@ export default class FeatureDiscovery extends React.PureComponent<Props, State>
         const canRequestCloudFreeTrial = isCloud && !isCloudTrial && !hadPrevCloudTrial && !isPaidSubscription;
 
         // by default we assume is not cloud, so the cta button is Start Trial (which will request a trial license)
-        let primaryMessage = (
-            <FormattedMessage
-                id='admin.ldap_feature_discovery.call_to_action.primary'
-                defaultMessage='Start trial'
-            />
-        );
-        let ctaButtonFunction: (e: React.MouseEvent) => void = this.requestLicense;
-
-        // then if it is cloud, and this account already had a free trial, then the cta button must be Upgrade now
-        if (isCloud && hadPrevCloudTrial) {
-            ctaButtonFunction = this.openUpgradeModal;
-            primaryMessage = (
-                <FormattedMessage
-                    id='admin.ldap_feature_discovery_cloud.call_to_action.primary'
-                    defaultMessage='Upgrade now'
-                />
-            );
-        }
-
-        // if all conditions are set for being able to request a cloud trial, then show the cta start cloud trial button
-        // otherwise use the previously defined conditions to elaborate the button
-        const ctaPrimaryButton = canRequestCloudFreeTrial ? (
-            <CloudStartTrialButton
-                message={Utils.localizeAndFormatMessage(
-                    'admin.ldap_feature_discovery.call_to_action.primary.cloudFree',
-                    'Try free for {trialLength} days',
-                    {trialLength: FREEMIUM_TO_ENTERPRISE_TRIAL_LENGTH_DAYS},
+        let ctaPrimaryButton = (
+            <StartTrialBtn
```

Comment:
I like using StartTriaBtn. If we do that, I think we need to add an optional prop to it, maybe called `trackingPage`, and then pass in `this.props.featureName`. Otherwise, we lose the tracking data of which feature led people to request a license. Looking at feature_discovery's [`requestLicense`](https://github.com/mattermost/mattermost-webapp/blob/master/components/admin_console/feature_discovery/feature_discovery.tsx#L92) method and the StartTrialBtn's [requestLicense](https://github.com/mattermost/mattermost-webapp/blob/master/components/learn_more_trial_modal/start_trial_btn.tsx#L68) function, that was the only meaninful difference in functionality I could see.

Separately, it looks like feature_discovery's `requestLicense` can be removed if we are using `StartTrialButton`.

Reply:
ok, I have to agree because the logic stablished for the team edition scenario is very complex and also because this is a difficult change to validate (really hard to do it locally, and at the time, it is imposible to do it with a spinwick server)... in conclusion, the risk is not worth the benefit. Thanks for pointing this out @neallred

Reply:
ok! done!